File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
  var response;
  if (input == input.toUpperCase()){
    return "Whoa, chill out!";
  }else{
    return punctuationChecker(input.slice(-1));
  }

  function punctuationChecker(punct){
    if (punct == "!" || punct == "."){
      return "Whatever."
    }else if(punct == "?"){
      return "Sure."
    }
  };
}

module.exports = Bob;

File: python/bob/bob.py
#
# Skeleton file for the Python "Bob" exercise.
#
REPLIES = [ "Whatever.", "Sure.", "Whoa, chill out!"]

def hey(what):
  char_phrase = strip_numbers(what)
  if char_phrase.upper() == char_phrase:
    return REPLIES[2]
  else:
    return check_punct(what[-1])

def strip_numbers(string):
  return ''.join(char for char in string if not char.isdigit())

def check_punct(punct):
  return {
    '.':REPLIES[0],
    '!':REPLIES[0],
    '?':REPLIES[1]
  }[punct]




